static/src/javascripts/projects/common/modules/gallery/lightbox.js

```diff
@@ -0,0 +1,626 @@
+                trailing: false,
+            })
+        );
+
+        bean.on(this.$swipeContainer[0], 'touchend', () => {
+            let direction;
+            if (Math.abs(dx) > threshold) {
+                direction = dx > threshold ? 1 : -1;
+            } else {
+                direction = 0;
+            }
+            dx = 0;
+
+            if (direction === 1) {
+                if (this.index > 1) {
+                    this.trigger('prev');
+                } else {
+                    this.trigger('reload');
+                }
+            } else if (direction === -1) {
+                if (this.index < this.$slides.length) {
+                    this.trigger('next');
+                } else {
+                    this.trigger('reload');
+                }
+            } else {
+                this.trigger('reload');
+            }
+        });
+    }
+
+    disableHover() {
+        this.$lightboxEl.removeClass('gallery-lightbox--hover');
+    }
+
+    trigger(event: string, data?: GalleryJson) {
+        this.fsm.trigger(event, data);
+    }
+
+    loadGalleryfromJson(galleryJson: GalleryJson, startIndex: number) {
+        this.index = startIndex;
+        if (this.galleryJson && galleryJson.id === this.galleryJson.id) {
+            this.trigger('open');
+        } else {
+            this.trigger('loadJson', galleryJson);
+        }
+    }
+
+    loadSurroundingImages(index: number, count: number) {
+        let imageContent;
+        let $img;
+
+        [-1, 0, 1]
+            .map(i => (index + i === 0 ? count - 1 : (index - 1 + i) % count))
+            .forEach(i => {
+                imageContent = this.images[i];
+                $img = bonzo(this.$images[i]);
+                if (!$img.attr('src')) {
+                    $img.parent().append(bonzo.create(loaderTpl));
+
+                    $img.attr('src', imageContent.src);
+                    $img.attr('srcset', imageContent.srcsets);
+                    $img.attr('sizes', imageContent.sizes);
+
+                    bean.one($img[0], 'load', () => {
+                        $('.js-loader').remove();
+                    });
+                }
+            });
+    }
+
+    translateContent(imgIndex: number, offset: number, duration: number) {
+        const px = -1 * (imgIndex - 1) * this.swipeContainerWidth;
+        const contentEl = this.$contentEl[0];
+
+        contentEl.style.webkitTransitionDuration = `${duration}ms`;
+        contentEl.style.mozTransitionDuration = `${duration}ms`;
+        contentEl.style.msTransitionDuration = `${duration}ms`;
+        contentEl.style.transitionDuration = `${duration}ms`;
+        contentEl.style.webkitTransform = `translate(${px +
+            offset}px,0) translateZ(0)`;
+        contentEl.style.mozTransform = `translate(${px + offset}px,0)`;
+        contentEl.style.msTransform = `translate(${px + offset}px,0)`;
+        contentEl.style.transform = `translate(${px +
+            offset}px,0) translateZ(0)`;
```

Comment:
Does flow complain of you do

```js
Object.assign(contentEl.style, {
 /* ... */
})
```

to avoid all the repetition?

Reply:
It does not! Great suggestions 👍